drivers/sensor/bmp581/bmp581.c

```diff
@@ -0,0 +1,577 @@
+/*
+ * Copyright (c) 2022 Badgerd Technologies B.V.
+ * Copyright (c) 2023 Metratec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bmp581.h"
+
+#include <math.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bmp581, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "BMP581 driver enabled without any devices"
+#endif
+
+static int power_up_check(const struct device *dev);
+static int get_nvm_status(uint8_t *nvm_status, const struct device *dev);
+static int get_interrupt_status(uint8_t *int_status, const struct device *dev);
+static int validate_chip_id(struct bmp581_data *drv);
+static int get_osr_odr_press_config(struct bmp581_osr_odr_press_config *osr_odr_press_cfg,
+				    const struct device *dev);
+static int set_osr_config(const struct sensor_value *osr, enum sensor_channel chan,
+			  const struct device *dev);
+static int set_odr_config(const struct sensor_value *odr, const struct device *dev);
+static int soft_reset(const struct device *dev);
+static int set_iir_config(const struct sensor_value *iir, const struct device *dev);
+static int get_power_mode(enum bmp5_powermode *powermode, const struct device *dev);
+static int set_power_mode(enum bmp5_powermode powermode, const struct device *dev);
+
+
+static int reg_read(uint8_t reg, uint8_t *data, uint16_t length, const struct device *dev)
+{
+	struct bmp581_config *conf = (struct bmp581_config *)dev->config;
+
+	return i2c_burst_read_dt(&conf->i2c, reg, data, length);
+}
+
+static int reg_write(uint8_t reg, const uint8_t *data, uint16_t length, const struct device *dev)
+{
+	struct bmp581_config *conf = (struct bmp581_config *)dev->config;
+
+	return i2c_burst_write_dt(&conf->i2c, reg, data, length);
+}
+
+static int set_power_mode(enum bmp5_powermode powermode, const struct device *dev)
+{
+	int ret = BMP5_OK;
+	uint8_t odr = 0;
+	enum bmp5_powermode current_powermode;
+
+	if (dev == NULL) {
```

Comment:
Most drivers don't check this. You can use `CHECKIF` to give the option to compile out the check

Reply:
replaced all those NULL pointer checks with `CHECKIF()`.